exec/cnex/lib/io.c

```diff
@@ -35,25 +60,13 @@ static FILE *check_file(TExecutor *exec, void *pf)
 }
 
 
-void io_fprint(TExecutor *exec)
-{
-    Cell *s = peek(exec->stack, 0);
-    void *pf = peek(exec->stack, 1)->other;
-
-    FILE *f = check_file(exec, pf);
-    fwrite(s->string->data, 1, s->string->length, f);
-    fputs("\n", f);
-
-    pop(exec->stack);
-    pop(exec->stack);
-}
-
 void io_close(TExecutor *exec)
 {
-    void *ppf = peek(exec->stack, 0)->address->other;
+    void *ppf = peek(exec->stack, 0)->object->ptr;
 
     FILE *f = check_file(exec, ppf);
     fclose(f);
+    f = NULL;
```

Comment:
This isn't going to do the trick. You're clearing the local variable `f`, which isn't going to mean much. What you need to clear is the `ptr` member so further file operations can fail cleanly.

Reply:
Fixed.